Feature/ab#89 legal guardian email change validation

vyruz1986 opened the pull request.

vyruz1986 committed:
fix(UserService): Check for duplicate email when updating user

Closes AB#89

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#89-legal-guardian-email-change-validation

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
fix(UserService,Frontend): Allow dupe emails for careusers.

Also added snackbar errors and confirmation messages in all relevant
locations

Closes AB#89

vyruz1986 committed:
fix(UserService, Frontend): Add extra email check on adduser.

Also implemented handling of error objects consisting of multiple errors
from backend

Closes AB#89

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#89-legal-guardian-email-change-validation

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

BerendWouters commented on Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts:
```diff
@@ -9,10 +9,9 @@ import {
    MatDialog,
-   MAT_DATE_FORMATS,
+   MatSnackBar,
 } from '@angular/material';
```
Ooh, painful. We will need this for our date formatting

BerendWouters commented on Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts:
```diff
@@ -68,10 +68,22 @@ export class OverviewComponent implements OnInit, AfterViewInit {
 
       dialogRef.componentInstance.submitEvent.subscribe((result: CareUser) => {
          // Update the Careuser
-         this.careUserService.updateUser(result).subscribe(res => {
```
Remember our discussion regarding javascript and by reference passing? It seems that javascript does this for objects and arrays, and only (obviously) for primitive types by value. So yes, updating the row values (even in a popup), means actually updating the row.

vyruz1986 replied:
It has been fixed

BerendWouters commented on Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts:
```diff
@@ -123,4 +145,24 @@ export class OverviewComponent implements OnInit, AfterViewInit {
+            messages.push(err[k]);
+         }
+         this.snackBar.open(
+            `⚠ Er zijn fouten opgetreden bij het opslagen:\n${messages.join(
```
Opslagen should be "opslaan"

vyruz1986 replied:
Fixed

BerendWouters commented on Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-overview/singerevent-overview.component.ts:
```diff
@@ -152,4 +176,24 @@ export class SingerEventOverviewComponent implements OnInit, AfterViewInit {
          .subscribe();
    }
+
+   handleApiError(err: any) {
```
Fancy!

BerendWouters commented on Singer.API/ClientApp/package.json:
```diff
@@ -52,11 +52,11 @@
       "typescript": "~3.2.4"
    },
    "optionalDependencies": {
+      "node-sass": "^4.9.3",
```
What happened here? It causes the build to break.

vyruz1986 replied:
Thanks for fixing :)

vyruz1986 replied:
Thanks for fixing :)

BerendWouters committed:
fix dependencies

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#89-legal-guardian-email-change-validation

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
fix: Fixed passing row as reference to details popup

The rows are now being dereferenced, so that we don't have to reload
them in case the API refuses an update

Closes AB#89
